### ISSUE TITLE
Mac (with M1 chipset) problem with VirtualBox

### ISSUE THREAD

**Issue by kevin-w-du** [seed-labs/seed-labs#54]
This is an issue raised by one of the students in my class. 
For those using Mac with M1 chipset, VMware and VirtualBox won't work because M1 uses ARM, and the SEED VM uses x86. After a lot of searching, the student reported that parallels (free for MAC m1 users) works.  Here is the link: 
https://b2b.parallels.com/apple-silicon/?gclid=Cj0KCQiApY6BBhCsARIsAOI_GjaBYUxRXDG3pXcZQcDQJDF0mukuABeqz5fI_xxsc-6g1LWpGOqTLekaArbAEALw_wcB.  We will see how it goes for this semester, and then revise the VM manual accordingly.

**Comment by Jacob953** [comment]
Although parallels works, there are still some problems.
When I run ` docker-compose build ` in /Labsetup which provided in official website, the error is as follows
`
Step 8/11 : RUN  a2ensite apache_elgg.conf
 ---> [Warning] The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
 ---> Running in 34e9f647cef2
standard_init_linux.go:228: exec user process caused: exec format error
The command '/bin/sh -c a2ensite apache_elgg.conf' returned a non-zero code: 1
ERROR: Service 'elgg' failed to build : Build failed
`
Is there any correct package for linux/amd64?

**Comment by kevin-w-du** [comment]
Before this problem is resolved, we will refer students to the cloud if their machine is based on M1. We don't have solution to this problem at this moment, and hopefully somebody in the community can help resolve this problem. I don't even own a M1 machine.

**Comment by Tweeks-va** [comment]
Yes.. the M1 problem is something that Virginia Tech is struggling with right now too. For now they're just trying to push back mass adaoption of M1 laptops.. and the cloud (with a pure web or RDP front end) makes for a nice solution.

**Comment by RingoHanabi** [comment]
Have we managed to solve this issue on M1/M2 Mac at this stage, or cloud is still the only solution?

**Comment by kevin-w-du** [comment]
VirtualBox is working on this, and has released the support for M1/M2, but it is not stable yet. Give it one more year, the problem should be fixed. We will reevaluate the situation next year and see whether VirtualBox is ready for us. At this point, Cloud is still the best solution.

**Comment by bdlowery** [comment]
2 years later... 🦗🦗🦗

**Comment by RingoHanabi** [comment]
> 2 years later, still crashes on boot even with the m1/m2 compatible Virtual box. Any fixes coming?

We've came across a solution to convert the VM file into .qcow2 file using qemu and it can be booted in UTM and qemu-simulation in a M2 MBP. But virtual box is still not workable for us

**Comment by HemanthDavuluri** [comment]
@RingoHanabi  can you provide documentation for that process? converting VM file into .qcow2

**Comment by RingoHanabi** [comment]
Here's a guide we used in Monash University's software security unit. It's a emulation approach so the performance is very bad, but is working. Hope it helps
https://docs.google.com/document/d/1rtzQkGgSEQt0-EqdPppN_HjJpWWbcG2hFDr219NrkJo/edit?usp=sharing

**Comment by kevin-w-du** [comment]
Thanks for the information. A student of mine is currently working with me to port the SEED labs 2.0 to the Apple Silicon machine. You can see our progress and partial results from this link: https://github.com/seed-labs/seed-labs/tree/master/lab-setup/apple-arm. We have decided to use VMWare Fusion player (free) to run an ARM version of Ubuntu 22.04 (instruction can be found from this link). It runs very well. We are now testing each of the SEED labs to see how it works.  Most labs should be fine (still doing the testing), but some labs will definitely break (e.g., the buffer overflow lab, because it uses x86 machine code in the exploit). We will have to redesign those labs for the ARM computers. So stay tuned.
